gitpod-io/gitpod
# Implement a new Team Billing where Owners can conveniently manage a paid plan for their Team

## Description

Allow teams to easily purchase and manage a Team Plan directly from their team's Settings.

TODO:
- [x] Implement a new Team Plan shape that's directly linked to / owned by a team: https://github.com/gitpod-io/gitpod/pull/9655
- [x] Allow subscribing via Chargebee directly from your team Settings
- [x] Create a "synthetic" user subscription when you're a member of a paid Team Plan
- [x] Transparently update the Team Plan quantity in Chargebee when users join or leave a team
- [x] Pro-rate charge for new members joining in the middle of a billing cycle
- [x] Allow managing your Billing in team Settings via a "Team Chargebee Portal"
- [x] Align the new UI with #8352 

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/7759

## How to test


1. Create a Team
2. Invite other team members
3. Go to Team Settings > Plans
3. Buy a Team Plan, verify billing amount matches member count, verify all members are indeed upgraded
4. Add/remove members, verify that the Team Plan billing amount is updated accordingly
5. Verify that cancelling, re-upgrading, deleting the team all work as expected

## Release Notes

```release-note
Implement a new Team Billing where Team Owners can conveniently manage a paid plan for their Team
```

## Documentation


- [ ] /werft without-vm=true
- [ ] /werft with-helm=true
